gamemode/modules/commands/sv_commands.lua

```diff
@@ -0,0 +1,314 @@
+util.AddNetworkString( "RVR_Commands_runConsoleCommand" )
+
+RVR.Commands = RVR.Commands or {}
+local commands = RVR.Commands
+
+commands.commands = {}
+commands.types = {}
+
+function commands.addType( name, func )
+    if commands.types[name] then
+        error( "A type with the same name (" .. name .. ") already exists" )
+    end
+
+    commands.types[name] = func
+end
+
+commands.addType( "int", function( arg )
+    local num = tonumber( arg )
+
+    if not num or num % 1 ~= 0 then
+        return nil, "Invalid integer: " .. arg
+    end
+
+    return num
+end )
+
+commands.addType( "float", function( arg )
+    local num = tonumber( arg )
+
+    if not num then
+        return "Invalid float: " .. arg
+    end
+
+    return num
+end )
+
+local booleanValues = {
+    ["enabled"]  = true,
+    ["enable"]   = true,
+    ["true"]     = true,
+    ["yes"]      = true,
+    ["1"]        = true,
+
+    ["disabled"] = false,
+    ["disable"]  = false,
+    ["false"]    = false,
+    ["no"]       = false,
+    ["0"]        = false
+}
+
+commands.addType( "bool", function( arg )
+    if booleanValues[arg] ~= nil then
+        return booleanValues[arg]
+    end
+
+    return nil, "Invalid boolean: " .. arg
+end )
+
+commands.addType( "string", function( arg )
+    return arg
+end )
+
+commands.addType( "player", function( arg, ply )
+    if arg == "@" then return arg end
```

Comment:
This is more what I initially intended `@` to do. I think we can ignore how a `*` would act for now, so we can just get this merged.
```suggestion
    if arg == "@" then
        local aimEnt = ply:GetEyeTrace().Entity
        if aimEnt and type( aimEnt ) == "Player" then return aimEnt end
        return nil, "Invalid aim entity"
    end
```